src/Feed/Feed.js

```diff
@@ -1,50 +1,67 @@
-import { GET_ALL_POSTS } from '../requests.js'
-import React, { useState } from 'react';
-import '../Scss/base.scss';
+import { addRing, removeAllRings } from '../mapActions.js';
+import { cycleIndex } from '../helperFx.js'
+import { GET_ALL_POSTS } from '../requests.js';
 import { gql, useQuery } from '@apollo/client';
 import { Post } from '../Post/Post.js';
+import React, { useState } from 'react';
+import '../Scss/base.scss';
 import prev from '../images/prev-mixed.png'
 import next from '../images/next-mixed.png'
 
-
 export default function Feed(props) {
-    let [postIndex, setPostIndex] = useState(0)
     let {loading, error, data } = useQuery(GET_ALL_POSTS);
+    let [earthMapIsLoaded, setEarthMapIsLoaded] = useState(window.earthMap !== undefined)
+    let [needNewRing, setNeedNewRing] = useState(true)
+    let [postIndex, setPostIndex] = useState(0)
 
     if (loading) return <h1>LOADING POSTS...</h1>
     if (error) return <h1>Hmm... something went wrong.</h1>
 
-    function addToIndex(num) {
-      let arrLength = data.posts.length
-      let total = num + postIndex
-      let index = ((total % arrLength) + arrLength) % arrLength
-      return index
-    }
-
-    let { content, createdAt, id, ringMinMax } = data.posts[postIndex]
+    let { content, createdAt, id, latitude, longitude, ringMinMax } = data.posts[postIndex]
     let [min, max] = ringMinMax.slice(1, -1).split(', ').map( char => +char )
 
+    if (needNewRing && earthMapIsLoaded) {
+      if (window.earthRings.length > 0) removeAllRings()
+      let center = {lat: () => latitude, lng: () => longitude}
+      let newRing = addRing( center, min, max )
```

Comment:
Nice!

Reply:
Oooh, remind me to take that conditional out. It is redundant any more, leftover from a previous design implementation.